src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -1135,7 +1135,7 @@ private void CheckExpectedType(JsonTokenType expected, JsonTokenType actual)
             JsonReaderOptions readerOptions,
             string paramName = null)
         {
-            if (readerOptions.CommentHandling == JsonCommentHandling.Allow)
+            if (!JsonHelpers.IsCommentHandlingSupported(readerOptions.CommentHandling, JsonCommentHandling.Skip))
```

Comment:
I didn't add "invalid" enum tests to `JsonDocument` since there was no way to get to such a state. I still decided to use the helper for consistency.